src/envoy/auth/config.cc

```diff
@@ -0,0 +1,138 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "config.h"
+
+#include "common/filesystem/filesystem_impl.h"
+#include "common/json/json_loader.h"
+#include "envoy/json/json_object.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "rapidjson/document.h"
+
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace Envoy {
+namespace Http {
+namespace Auth {
+
+void AsyncClientCallbacks::onSuccess(MessagePtr &&response) {
+  auto len = response->body()->length();
+  std::string body(static_cast<char *>(response->body()->linearize(len)), len);
+  auto status = std::string(response->headers().Status()->value().c_str());
+  if (status == "200") {
+    cb_(true, body);
+  } else {
+    cb_(false, "");
```

Comment:
can we log status code if not 200. for easy debugging

Reply:
we should use ENVOY_LOG to log?

Reply:
done